packages/interface/src/utils/fetchProjects.ts

```diff
@@ -0,0 +1,103 @@
+import { config } from "~/config";
+
+import { createCachedFetch } from "./fetch";
+import { IRecipient } from "./types";
+
+const cachedFetch = createCachedFetch({ ttl: 1000 });
+
+export interface GraphQLResponse {
+  data?: {
+    recipients: IRecipient[];
+  };
+}
+
+// query to fetch all approved projects
+const ApprovedProjects = `
+  query Recipients($registryAddress: String) {
+    recipients (where:{ deleted:false, initialized: true, registry: $registryAddress }) {
+      id
+      payout
+      metadataUrl
+      index
+      initialized
+      registry {
+        id
+      }
+    }
+  }`;
+
+const Projects = `
+  query Recipients($registryAddress: String) {
+    recipients (where:{ deleted:false, registry: $registryAddress }) {
+      id
+      payout
+      metadataUrl
+      index
+      initialized
+      registry {
+        id
+      }
+    }
+  }`;
+
+/**
+ * Fetch all projects
+ *
+ * @returns the projects
+ */
+export async function fetchProjects(registryAddress: string): Promise<IRecipient[]> {
+  const response = await cachedFetch<{ recipients: IRecipient[] }>(config.maciSubgraphUrl, {
+    method: "POST",
+    body: JSON.stringify({
+      query: Projects,
+      variables: { registryAddress },
+    }),
+  }).then((resp: GraphQLResponse) => resp.data?.recipients);
+
+  const recipients = response?.map((request) => ({
+    id: request.id,
+    metadataUrl: request.metadataUrl,
+    payout: request.payout,
+    initialized: request.initialized,
+    index: request.index,
+  }));
+
+  return recipients ?? [];
+}
+
+/**
+ * Fetch all approved projects
+ *
+ * @returns the projects
+ */
+export async function fetchApprovedProjects(registryAddress: string): Promise<IRecipient[]> {
+  const response = await cachedFetch<{ recipients: IRecipient[] }>(config.maciSubgraphUrl, {
+    method: "POST",
+    body: JSON.stringify({
+      query: ApprovedProjects,
+      variables: { registryAddress },
+    }),
+  }).then((resp: GraphQLResponse) => resp.data?.recipients);
+
+  const recipients = response?.map((request) => ({
+    id: request.id,
+    metadataUrl: request.metadataUrl,
+    payout: request.payout,
+    initialized: request.initialized,
+    index: request.index,
+  }));
+
+  return recipients ?? [];
+}
+
+/**
+ * Fetch a project by ID
+ *
+ * @param registryAddress - the registry ID
+ * @param id - the project ID
+ * @returns the project
+ */
+export async function fetchProjectById(registryAddress: string, id: string): Promise<IRecipient | undefined> {
+  const projects = await fetchProjects(registryAddress);
+  return projects.find((project) => project.id === id);
```

Comment:
Is it possible to fetch project with this id so we don't need to find all of them after fetching?

Reply:
it's actually not even used so might just remove